src/pain/001/SEPACreditPaymentInitiation.ts

```diff
@@ -0,0 +1,148 @@
+import { create } from "xmlbuilder2";
+import { Account, Agent, BICAgent, IBANAccount, Party, SEPACreditPaymentInstruction } from "../../lib/types";
+import { PaymentInitiation } from "./ISO20022PaymentInitiation"
+import { sanitize } from "../../utils/format";
+import Dinero from 'dinero.js';
+import { v4 as uuidv4 } from 'uuid';
+
+type AtLeastOne<T> = [T, ...T[]];
+
+export interface SEPACreditPaymentInitiationConfig {
+  initiatingParty: Party;
+  paymentInstructions: AtLeastOne<SEPACreditPaymentInstruction>;
+  messageId?: string;
+  creationDate?: Date;
+}
+
+export class SEPACreditPaymentInitiation extends PaymentInitiation {
+  private initiatingParty: Party;
+  private messageId: string;
+  private creationDate: Date;
+  private paymentInstructions: AtLeastOne<SEPACreditPaymentInstruction>;
+  private paymentSum: string;
+
+  constructor(config: SEPACreditPaymentInitiationConfig) {
+    super();
+    this.initiatingParty = config.initiatingParty;
+    this.paymentInstructions = config.paymentInstructions.map(this.reifyPaymentInstruction) as AtLeastOne<SEPACreditPaymentInstruction>;
+    this.messageId = config.messageId || uuidv4().replace(/-/g, '');
+    this.creationDate = config.creationDate || new Date();
+    this.validate();
+    this.paymentSum = this.sumPaymentInstructions(this.paymentInstructions as AtLeastOne<SEPACreditPaymentInstruction>);
+  }
+
+  private reifyPaymentInstruction(instruction: SEPACreditPaymentInstruction): SEPACreditPaymentInstruction {
+    return {
+      ...instruction,
+      dinero: Dinero({ amount: instruction.amount, currency: instruction.currency }),
+    };
+  }
```

Comment:
What is the purpose of this method? Can we create Dineros on the fly when we need to, instead of saving it?